File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = Integer.MAX_VALUE;

        for (int length = 1; length <= s.length() / 2 + 1; length++) {
            StringBuilder sb = new StringBuilder();
            String pre = "";
            int count = 1;
            for (int i = 0; i < s.length(); i += length) {
                String sub = (s.length() - i < length) ? s.substring(i) : s.substring(i, i + length);

                if (sub.equals(pre)) {
                    count++;
                } else {
                    if(count < 2) {
                        sb.append(pre);
                    }else {
                        sb.append(count).append(pre);
                    }
                    count = 1;
                }
                pre = sub;
            }

            if(count < 2) {
                sb.append(pre);
            }else {
                sb.append(count).append(pre);
            }
            answer = Math.min(sb.length(), answer);
        }

        return answer;
    }
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = s.length();
    int maxLength = s.length() / 2;

    for (int length = 1; length <= maxLength; ++length)
    {
        string str = s;
        string result = "";
        string last = "";
        int count = 0;
        for (int i = 0; i < s.length(); i += length)
        {
            string cut = str.substr(i, length);

            if (last == cut)
            {
                ++count;
            }
            else
            {

                if (count > 1)
                    result += to_string(count) + last;
                else if (count == 1)
                    result += last;

                last = cut;
                count = 1;
            }
        }

        if (count > 1)
            result += to_string(count) + last;
        else if (count == 1)
            result += last;

        if (result.length() < answer)
        {
            answer = result.length();
        }
    }

    return answer;
}